opentracing-util/src/main/java/io/opentracing/util/ThreadLocalActiveSpanSource.java

```diff
@@ -36,7 +34,12 @@ public ThreadLocalActiveSpan activeSpan() {
 
     @Override
     public ActiveSpan makeActive(Span span) {
-        return new ThreadLocalActiveSpan(this, span, new AtomicInteger(1));
+        return new ThreadLocalActiveSpan(this, span, new AutoFinisher());
```

Comment:
I think the refcounting should/must be a per-instrumentation-site decision. When it's appropriate, it's great. When it's not, it's not.

@yurishkuro 

> those people who asked for no refcounting / auto-finish behavior, are they implementing a Tracer or doing instrumentation of some framework?

Both, I think... though I'm more concerned with the latter all things being equal (as it's a larger population of people).

We will need to document at the API level what it means when no observer is provided (i.e., if it implies no-auto-finish or yes-auto-finish). I guess my current proposal is that the default should be yes-auto-finish, but that is definitely debatable and is probably the most interesting choice to make in this entire PR.

Very interested in the opinions of the folks I've cc'd on this PR (or any others).